commonTools/build_stats/BuildStatsWrappers.cmake

```diff
@@ -8,6 +8,92 @@ set(BUILD_STATS_SRC_DIR "${CMAKE_CURRENT_LIST_DIR}")
 
 set(BUILD_STATS_CSV_FILE "${${PROJECT_NAME}_BINARY_DIR}/build_stats.csv")
 
+function(build_stats_find_time)
+
+  # let the user provide BUILD_STATS_TIME_CMD
+  if (BUILD_STATS_TIME_CMD)
+    message("-- BUILD_STATS_TIME_CMD=${BUILD_STATS_TIME_CMD}")
+    set(GNU_TIME_EXE "${BUILD_STATS_TIME_CMD}")
+  else()
+    find_program(GNU_TIME_EXE "time" HINTS "/usr/bin")
+    if(GNU_TIME_EXE)
+      message("-- Found time at ${GNU_TIME_EXE}")
+    else()
+      message("-- GNU time NOT found")
+      message("-- Install GNU time and/or set BUILD_STATS_TIME_CMD=/path/to/time")
+      return()
+    endif()
+  endif()
+
+  # this should ideally call the python script and request the fields
+  # to test, add 'badflag' or some other nonsense
+  SET(GNU_TIME_POSSIBLE_FIELDS "e;M;K;D;X;F;R;W;w;c;S;U;P;I;O;r;s;k;x")
+  SET(GNU_TIME_SUPPORTED_FIELDS "")
+  
+  #  should ideally ask for the dtypes or suitable regexes to vet them
+  foreach(flag ${GNU_TIME_POSSIBLE_FIELDS})
+    message(DEBUG "----------------------")
+    message(DEBUG "Time: Testing field ${flag}")
+    # the output from time goes to stderr, the programs output to stdout
+    set(GNU_TIME_RC "-1")
+    set(GNU_TIME_OUTPUT "")
+    execute_process(COMMAND "${GNU_TIME_EXE}"
+                              "--format=%${flag}" "true"
+                    # this is useless - we run a noop command
+                    RESULT_VARIABLE GNU_TIME_RC
+                    # capture stderr
+                    ERROR_VARIABLE GNU_TIME_OUTPUT
+                    )
+    # if this fails, then something is broken on the system.
+    # the checks after will likely fail, because they expect a predefined
+    # format for stderr text
+    if(GNU_TIME_RC MATCHES "0")
+      #message("Time invocation error returned `${GNU_TIME_RC}` but expected `0`")
+    else()
+      message(DEBUG "Time invocation error returned `${GNU_TIME_RC}` but expected `0`")
+      message("-- GNU_TIME_EXE=${GNU_TIME_EXE} does not work")
+      message("-- Set BUILD_STATS_TIME_CMD if /usr/bin/time is invalid")
+      unset(BUILD_STATS_TIME_CMD CACHE)
+      return()
+    endif()
+
+    # this isn't ideal, we match the failure mode, but we should match valid output
+    # That is harder to do, as we would need to determine expected format for each
+    # command (e.g., dtype or a suitable regex from the python script)
+    # A good todo would be to have the pythong script report headeres it will
+    # try and a valid expected format regex, then we test against that here.
+    if("${GNU_TIME_OUTPUT}" MATCHES "^?${flag}.*")
+      message(DEBUG "Time does not support Field: ${flag}")
+      # not ideal, but let's quit if we don't find the fields...
+      # a better solution would be to disable the field, and only
+      # error out if we can't find some required set of fields.
```

Comment:
I'd need to double check what Time does.  I know if you give strings (that aren't format strings, like `%e`, it just passes that back to you).  E.g., CSV output is just a format string of these fields `--format='%e , %E`  The commas (and spaces) are preserved.

If you pass an unsupported field, you generally will get `?<name>` e.g.,
```
[jjellio@desktop ~]$ /usr/bin/time --format='%e, %NOPE' hostname
desktop.sandia.gov
0.00, ?NOPE
```

I'd like to ensure that I don't pass fields that aren't supported, as this will cause the `gather` commands to fail

Reply:
> I'd like to ensure that I don't pass fields that aren't supported, as this will cause the gather commands to fail

The upgraded `gather_build_stats.py` tool is pretty robust.  I suspect that as long a something get written into a given row that it will allow it.